engine/src/main/java/org/terasology/rendering/dag/FinalPostProcessingNode.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.Display;
+import org.lwjgl.opengl.GL13;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.config.RenderingDebugConfig;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.oculusVr.OculusVrHelper;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+import org.terasology.rendering.opengl.PostProcessor;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.rendering.world.WorldRenderer.RenderingStage;
+import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
+import static org.lwjgl.opengl.GL11.glClear;
+import static org.terasology.rendering.opengl.OpenGLUtils.bindDisplay;
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+import static org.terasology.rendering.opengl.OpenGLUtils.setViewportToSizeOf;
+
+/**
+ * TODO: Add diagram of this node
+ * TODO: Break into two different nodes
+ */
+public class FinalPostProcessingNode implements Node {
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private Config config;
+
+    @In
+    private PostProcessor postProcessor;
+
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
+    private RenderingDebugConfig renderingDebugConfig;
+    private RenderingConfig renderingConfig;
+
+    private FBO.Dimensions fullScale;
+
+    private Material finalPost;
+    private Material debug;
+    private Material ocDistortion;
+
+    private FBO sceneFinal;
+    private FBO ocUndistorted;
+    private FBO sceneOpaque;
+
+    @Override
+    public void initialise() {
+        renderingConfig = config.getRendering();
+        renderingDebugConfig = renderingConfig.getDebug();
+
+        ocDistortion = worldRenderer.getMaterial("engine:prog.ocDistortion");
+        finalPost = worldRenderer.getMaterial("engine:prog.post"); // TODO: rename shader to finalPost
+        debug = worldRenderer.getMaterial("engine:prog.debug");
```

Comment:
Add //TODO: rethink debug strategy in light of DAG-based architecture - it should be possible to inspect the output of any rendering node. I'll also write an issue in your fork on this.